src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -60,6 +60,14 @@ using std::auto_ptr;
 
 /* ****************************************************************************
 *
+* OtisTreatFunction - callback signature for treatOntimeintervalSubscriptions
+*/
+typedef void (*OtisTreatFunction)(std::string tenant, BSONObj* bobjP);
```

Comment:
The answers to that question say:

> Use reference wherever you can, pointers wherever you must.
> Avoid pointers until you can't.
> So the rule of thumb is to use pointers only if there is no other choice. 

In this case, is there any reason that force to use a pointer? Moreover, given that the BSON object passed to the function is not modified (as far as I remember) in the recovery/destroy logic, probably a `conts BSONObj&` could be used.